File: Days 001 - 010/Day 10/JobScheduler.cpp
#include <chrono>
#include <iostream>
#include <thread>
#include <type_traits>

template<typename Func, typename... Args, typename = std::enable_if_t<std::is_invocable_v<Func, Args...>>>
void ScheduleJob(Func function, unsigned int callbackMilliseconds, Args... params)
{
	std::this_thread::sleep_for(std::chrono::milliseconds(callbackMilliseconds));

	function(params...);
}

void JobA()
{
	std::cout << "Job A has been completed." << "\n";
}

void JobB(int a, int b)
{
	std::cout << "Job B completed with result of " << (a + b) << "." << "\n";
}

void JobC(const char* text, float a, float b)
{
	std::cout << text << "\n";
	std::cout << "Job C completed with result of " << (a + b) << "." << "\n";
}

int main(int argc, char* argv[])
{
	ScheduleJob(JobA, 1500);
	ScheduleJob(JobB, 500, 7, 4);
	ScheduleJob(JobC, 3250, "This will print.", 2.5f, 9.0f);

	std::cin.get();

	return 0;
}

File: Days 001 - 010/Day 10/JobScheduler.cs
using System;
using System.Threading;

namespace DailyCodingProblem
{
	internal class JobScheduler
	{
		public static void ScheduleJob(Action function, int callback)
		{
			Thread.Sleep(callback);

			function();
		}
	}

	internal class Day10
	{
		private static int Main(string[] args)
		{
			JobScheduler.ScheduleJob(JobA, 1500);
			JobScheduler.ScheduleJob(JobB, 500);
			JobScheduler.ScheduleJob(JobC, 3250);

			Console.ReadLine();

			return 0;
		}

		private static void JobA()
		{
			Console.WriteLine("Job A completed.");
		}

		private static void JobB()
		{
			Console.WriteLine("Job B completed.");
		}

		private static void JobC()
		{
			Console.WriteLine("Job C completed.");
		}
	}
}